geode/src/main/scala-3/org/apache/pekko/stream/connectors/geode/impl/pdx/ObjectEncoder.scala

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.pekko.stream.connectors.geode.impl.pdx
+
+import org.apache.pekko.annotation.InternalApi
+
+@InternalApi
+private[pekko] trait ObjectEncoder {}
```

Comment:
This is still tbd

Reply:
Do you think, this is harder to do than the decoder part? Or did you not yet get to it?

Reply:
I did not get to it. I am not expecting any reason for it to be harder.

The main problem here personally is that my understanding of Shapeless is pretty surface level and Scala 3 has completely redone how metaprogramming works (as you can see they basically put shapeless into the core language with different abstractions).

Reply:
If you want to take a shot at it please go ahead, feel free to add commits onto my PR